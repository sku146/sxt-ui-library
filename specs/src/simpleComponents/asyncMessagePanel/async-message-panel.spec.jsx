import React from 'react';
import renderer from 'react-test-renderer';
import AsyncMessagePanel from 'src/simpleComponents/asyncMessagePanel/async-message-panel';

describe("Async Message Panel", () => {

  let defaultProps = {
    name: 'testAsyncMessagePanel',
    msgType: 'error', // success, error, info, warning, plain(with white background), transparent
    pointPosition: 'left',
    pointing: true,
    isRequested: true,
  };

  const RendererComponent = (props) => {
    return renderer.create(<AsyncMessagePanel {...props}>
      <div>test</div>
    </AsyncMessagePanel>);
  };

  it('component should render as expected', ()=> {
    const component = RendererComponent(defaultProps);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('component should render as expected with isRequested FALSE', ()=> {
    defaultProps.isRequested = false;
    const component = RendererComponent(defaultProps);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
