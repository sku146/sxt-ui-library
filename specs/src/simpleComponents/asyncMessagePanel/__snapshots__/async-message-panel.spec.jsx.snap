// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async Message Panel component should render as expected 1`] = `
<div
  className="ui pointing basic label message-error position-left"
  data-bdd-selector="testAsyncMessagePanelErrorMsg"
  onClick={[Function]}
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui inverted loader"
        />
      </div>
    </div>
  </div>
  <div>
    test
  </div>
</div>
`;

exports[`Async Message Panel component should render as expected with isRequested FALSE 1`] = `
<div
  className="ui pointing basic label message-error position-left"
  data-bdd-selector="testAsyncMessagePanelErrorMsg"
  onClick={[Function]}
>
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui inverted loader"
        />
      </div>
    </div>
  </div>
  <div>
    test
  </div>
</div>
`;
